units/src/fee_rate.rs

```diff
@@ -40,29 +40,22 @@ impl FeeRate {
     /// Minimum fee rate required to broadcast a transaction.
     ///
     /// The value matches the default Bitcoin Core policy at the time of library release.
-    pub const BROADCAST_MIN: FeeRate = FeeRate::from_sat_per_vb_unchecked(1);
+    pub const BROADCAST_MIN: FeeRate = FeeRate::from_sat_per_vb(1);
 
     /// Fee rate used to compute dust amount.
-    pub const DUST: FeeRate = FeeRate::from_sat_per_vb_unchecked(3);
+    pub const DUST: FeeRate = FeeRate::from_sat_per_vb(3);
 
     /// Constructs [`FeeRate`] from satoshis per 1000 weight units.
     pub const fn from_sat_per_kwu(sat_kwu: u64) -> Self { FeeRate(sat_kwu) }
 
     /// Constructs [`FeeRate`] from satoshis per virtual bytes.
-    ///
-    /// # Errors
-    ///
-    /// Returns [`None`] on arithmetic overflow.
-    pub fn from_sat_per_vb(sat_vb: u64) -> Option<Self> {
+    pub const fn from_sat_per_vb(sat_vb: u32) -> Self {
```

Comment:
This is #3166 ;)

I think it'd make sense to have a method that takes `u32` named `from_u32_sat_per_vb`. For consistency sake I don't think we should remove the old one.

Reply:
> This is https://github.com/rust-bitcoin/rust-bitcoin/issues/3166 ;)

haha I also thought this.  Of all the things to commit to change the api, I wouldn't think this is worhwhile.

Reply:
There really is a hard trade off between "improve the lib" and "don't break anything".

Reply:
If there was `u62` I'd just go for the API change but for `u32` it's awkward.